pkg/idempotent/redis/redis.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+// redis is a simple inmemory Recorder implementation. This implementation
+// is intended for simple usecases (local dev) and not production workloads.
+package redis
+
+import (
+	"context"
+
+	redis "github.com/gomodule/redigo/redis"
+)
+
+var (
+	// Used for Redis value
+	defaultAddress = "localhost:6379"
+	defaultTimeout = 86400 //The default timeout should be 24 hours.
+	defaultValue   = struct{}{}
+)
+
+func New() *Redis {
+	return &Redis{}
+}
+
+type Redis struct {
+}
+
+func (r *Redis) SeenBefore(key string) (bool, context.Context) {
+	var ctx context.Context
+	conn, err := redis.Dial("tcp", defaultAddress)
+	if err != nil {
+		ctx = context.WithValue(ctx, "redis dial error", err)
+	}
+	defer conn.Close()
+	seen, err := redis.Bool(conn.Do("EXISTS", key))
```

Comment:
Can we wrap the `EXISTS` and `SETEX` commands in a `WATCH` transaction? That way we have atomic compare-and-swap.

https://redis.io/topics/transactions#cas